### DISCUSSION TITLE
[Section 5.26 -  Criando um site estático] - Buckets

### DISCUSSION THREAD

**Discussion by jeffersonfragoso** [chgasparoto/curso-aws-com-terraform#146] [Q&A] [answered]
Estou recebendo o seguinte erro:

` Plan: 32 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + website-url = (known after apply)
╷
│ Error: Invalid count argument
│ 
│   on .terraform/modules/website/main.tf line 36, in resource "aws_s3_bucket_policy" "this":
│   36:   count = var.policy != "" ? 1 : 0
│ 
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many
│ instances will be created. To work around this, use the -target argument to first apply only the resources that the count
│ depends on.`

Pelo visto não está aceitando passar a policy utilizando o 'rendered' do template.

**Comment by jeffersonfragoso** [comment, marked as answer]
Parece bug mesmo na passagem de valor para o módulo. O valor de  'rendered' é um (known after apply), porém não estava funcionando mesmo executando o terraform apply.

Depois fiz um teste isolado apenas carregando o arquivo e jogando o 'rendered' em um output e após rodar o apply esse teste deu certo

Por fim, depois desse teste voltei para o código da aula e tudo funcionou, como se estivesse pegando um tipo de cache...

**Reply by chgasparoto** [reply]
@jeffersonfragoso ok, obrigado pelo feedback!
